nixos/modules/services/networking/cgproxy.nix

```diff
@@ -0,0 +1,110 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cgproxy;
+  settingsFormat = pkgs.formats.json { };
+  settingsFile = settingsFormat.generate "config.json" cfg.settings;
+in
+{
+
+  options = {
+    services.cgproxy = {
+      enable = mkEnableOption (lib.mdDoc "cgproxy");
+
+      settings = mkOption {
+        description = lib.mdDoc ''
+          options priority:
+          program_noproxy > program_proxy > cgroup_noproxy > cgroup_proxy
+          enable_ipv6 = enable_ipv4 > enable_dns > enable_tcp = enable_udp
+          command cgproxy and cgnoproxy always have highest priority
+
+          Note: cgroup in configuration need to be exist, otherwise ignored
```

Comment:
Could we assert this?

Reply:
Nope, this is from upstream README and seems related to the `cgroup` at runtime, nix cannot handle this. I think this note may be unnecessary.